File: eMovieFinder/eMovieFinder.Model/Entities/CartItem.cs
using eMovieFinder.Model.Utilities;

namespace eMovieFinder.Model.Entities
{
    public class CartItem : TimeStampObjectViewModel
    {
        public int Id { get; set; }
        public int CartId { get; set; }
        public int MovieId { get; set; }
        public virtual Movie Movie { get; set; }
        public decimal FinalMoviePrice { get; set; }
        public string? FormattedFinalMoviePrice => FinalMoviePrice.ToString("F2");
    }
}

File: eMovieFinder-UI/emovie_finder_mobile/lib/src/models/entities/cartitem.dart
import 'package:json_annotation/json_annotation.dart';

import 'package:emovie_finder_mobile/src/models/utilities/time_stamp_object.dart';
import 'package:emovie_finder_mobile/src/models/entities/movie.dart';

part 'cartitem.g.dart';

@JsonSerializable()
class CartItem extends TimeStampObject {
  final int? id;
  final int? cartId;
  final int? movieId;
  final dynamic finalMoviePrice;
  final String? formattedFinalMoviePrice;
  final Movie? movie;

  CartItem({
    super.creationDate,
    super.modifiedDate,
    super.createdById,
    super.modifiedById,
    super.formattedCreationDate,
    super.formattedModifiedDate,
    this.id,
    this.cartId,
    this.movieId,
    this.finalMoviePrice,
    this.formattedFinalMoviePrice,
    this.movie
  });

  factory CartItem.fromJson(Map<String, dynamic> json) => _$CartItemFromJson(json);
  Map<String, dynamic> toJson() => _$CartItemToJson(this);
}